### PR TITLE
Add notifications for linked users

### PR DESCRIPTION
**Pull request by Hunterness**
Notifications introduced in https://github.com/neo-technology/neo4j/pull/25227 (merged, but behind feature flag). Feature flag removed in https://github.com/neo-technology/neo4j/pull/26923.

### CODE REVIEW

File: modules/ROOT/pages/notifications/all-notifications.adoc [hunk 1]
```diff
@@ -1241,6 +1241,113 @@ Suggestions for improvement::
...
 ====
 
+[#_neo_clientnotification_security_authprovidernotdefined]
+=== AuthProviderNotDefined
+
+.Notification details
+[cols="<1s,<4"]
+|===
+|Code
+m|Neo.ClientNotification.Security.AuthProviderNotDefined
```
Comment (Hunterness) [inline modules/ROOT/pages/notifications/all-notifications.adoc:1251]: This notification was covered by CIP-145.

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-status-codes-142.surge.sh

Updated pages:
- [Changes to status codes per Neo4j version](https://neo4j-docs-status-codes-142.surge.sh/status-codes/5/changelogs)
- [List of all notification codes](https://neo4j-docs-status-codes-142.surge.sh/status-codes/5/notifications/all-notifications)

File: modules/ROOT/pages/notifications/all-notifications.adoc [hunk 2]
```diff
@@ -1241,6 +1241,113 @@ Suggestions for improvement::
...
+====
+
+[#_neo_clientnotification_security_externalauthnotenabled]
+=== ExternalAuthNotEnabled
+
+.Notification details
+[cols="<1s,<4"]
+|===
+|Code
+m|Neo.ClientNotification.Security.ExternalAuthNotEnabled
```
Comment (Hunterness) [inline modules/ROOT/pages/notifications/all-notifications.adoc:1306]: This notification was an error in CIP-145 but was decided to be just a warning during implementation (to not lock out current external only users from being able to switch by first adding the new external only users to system before disabling their ways to log in)

File: modules/ROOT/pages/notifications/all-notifications.adoc [hunk 3]
```diff
@@ -1241,6 +1241,113 @@ Suggestions for improvement::
...
+====
+Command::
++
+[source, cypher]
+----
+CREATE USER foo SET AUTH 'exampleProvider' { SET ID 'idString' }
+----
+
+Description of the returned code::
+Use setting `dbms.security.require_local_user` to enable external auth.
```
Comment (Hunterness) [inline modules/ROOT/pages/notifications/all-notifications.adoc:1327]: do we need to have description on each individual example when it is in the notification table above and don't have any changing details?

File: modules/ROOT/pages/notifications/all-notifications.adoc [hunk 4]
```diff
@@ -1241,6 +1241,113 @@ Suggestions for improvement::
...
+
+.Notification details
+[cols="<1s,<4"]
+|===
+|Code
+m|Neo.ClientNotification.Security.ExternalAuthNotEnabled
+|Title
+a|External auth for user is not enabled.
+|Description
+a|Use setting `dbms.security.require_local_user` to enable external auth.
```
Comment (Hunterness) [inline modules/ROOT/pages/notifications/all-notifications.adoc:1310]: This setting is at the moment internal but will be made public when the feature flag is removed

File: modules/ROOT/pages/changelogs.adoc [hunk 5]
```diff
@@ -1,6 +1,16 @@
 :description: This page lists all changes to status codes per Neo4j version.
 = Changes to status codes per Neo4j version
 
+== Neo4j 5.21
```
Comment (Hunterness) [inline modules/ROOT/pages/changelogs.adoc:4]: remember to update this to correct version that this gets added in

File: modules/ROOT/pages/notifications/all-notifications.adoc (hunk 3 above)
Comment (renetapopova) [reply modules/ROOT/pages/notifications/all-notifications.adoc:1327]: No, if the title and descriptions are the same.

File: modules/ROOT/pages/notifications/all-notifications.adoc [hunk 6]
```diff
@@ -1241,6 +1241,113 @@ Suggestions for improvement::
...
+----
+CREATE USER foo SET AUTH 'exampleProvider' { SET ID 'idString' }
+----
+
+Description of the returned code::
+Use setting `dbms.security.require_local_user` to enable external auth.
+
+Suggestions for improvement::
+Enable linked users through the `dbms.security.require_local_user` setting.
+Until enabled the new external auth will be ignored and current external auth behaviours will continue to apply.
```
Comment (renetapopova) [inline modules/ROOT/pages/notifications/all-notifications.adoc:1331]:
```suggestion
Until enabled, the new external auth will be ignored, and current external auth behaviors will continue to apply.
```

File: modules/ROOT/pages/notifications/all-notifications.adoc [hunk 7]
```diff
@@ -1241,6 +1241,113 @@ Suggestions for improvement::
...
+----
+ALTER USER foo SET AUTH 'exampleProvider' { SET ID 'idString' }
+----
+
+Description of the returned code::
+Use setting `dbms.security.require_local_user` to enable external auth.
+
+Suggestions for improvement::
+Enable linked users through the `dbms.security.require_local_user` setting.
+Until enabled the new external auth will be ignored and current external auth behaviours will continue to apply.
```
Comment (renetapopova) [inline modules/ROOT/pages/notifications/all-notifications.adoc:1348]:
```suggestion
Until enabled, the new external auth will be ignored, and current external auth behaviors will continue to apply.
```

File: modules/ROOT/pages/notifications/all-notifications.adoc [hunk 8]
```diff
@@ -1241,6 +1241,113 @@ Suggestions for improvement::
...
+|Description
+a|The auth provider `<provider>` is not defined in the configuration.
+Verify that the spelling is correct or define `<provider>` in the configuration.
+|Severity
+m|INFORMATION
+|Category
+m|SECURITY
+|===
+
+.Create a user with an auth provider that isn't defined in config
```
Comment (renetapopova) [inline modules/ROOT/pages/notifications/all-notifications.adoc:1263]:
```suggestion
.Create a user with an auth provider that is not defined in the configuration
```

File: modules/ROOT/pages/notifications/all-notifications.adoc [hunk 9]
```diff
@@ -1241,6 +1241,113 @@ Suggestions for improvement::
...
+Description of the returned code::
+The auth provider `unknownProvider` is not defined in the configuration.
+Verify that the spelling is correct or define `unknownProvider` in the configuration.
+
+Suggestions for improvement::
+Make sure that the given provider is correct, or replace it if not.
+If it is correct, make sure to add it as a known auth provider in one or both of `dbms.security.authentication_providers` and `dbms.security.authorization_providers`.
+====
+
+.Alter a user to add an auth provider that isn't defined in config
```
Comment (renetapopova) [inline modules/ROOT/pages/notifications/all-notifications.adoc:1281]:
```suggestion
.Alter a user to add an auth provider that is not defined in the configuration
```

**Review by renetapopova — APPROVED** [review]
Some minor editorial suggestions. Otherwise, it looks good from my side.

File: modules/ROOT/pages/notifications/all-notifications.adoc [hunk 10]
```diff
@@ -1241,6 +1241,113 @@ Suggestions for improvement::
...
++
+[source, cypher]
+----
+CREATE USER foo SET AUTH 'exampleProvider' { SET ID 'idString' }
+----
+
+Description of the returned code::
+Use setting `dbms.security.require_local_user` to enable external auth.
+
+Suggestions for improvement::
```
Comment (Hunterness) [inline modules/ROOT/pages/notifications/all-notifications.adoc:1329]:
```suggestion
Command::
+
[source, cypher]
----
CREATE USER foo SET AUTH 'exampleProvider' { SET ID 'idString' }
----

Suggestions for improvement::
```

File: modules/ROOT/pages/notifications/all-notifications.adoc [hunk 11]
```diff
@@ -1241,6 +1241,113 @@ Suggestions for improvement::
...
++
+[source, cypher]
+----
+ALTER USER foo SET AUTH 'exampleProvider' { SET ID 'idString' }
+----
+
+Description of the returned code::
+Use setting `dbms.security.require_local_user` to enable external auth.
+
+Suggestions for improvement::
```
Comment (Hunterness) [inline modules/ROOT/pages/notifications/all-notifications.adoc:1346]:
```suggestion
Command::
+
[source, cypher]
----
ALTER USER foo SET AUTH 'exampleProvider' { SET ID 'idString' }
----

Suggestions for improvement::
```

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-status-codes-142.surge.sh

Updated pages:
- [Changes to status codes per Neo4j version](https://neo4j-docs-status-codes-142.surge.sh/status-codes/5/changelogs)
- [List of all notification codes](https://neo4j-docs-status-codes-142.surge.sh/status-codes/5/notifications/all-notifications)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-status-codes-142.surge.sh

Updated pages:
- [Changes to status codes per Neo4j version](https://neo4j-docs-status-codes-142.surge.sh/status-codes/5/changelogs)
- [List of all notification codes](https://neo4j-docs-status-codes-142.surge.sh/status-codes/5/notifications/all-notifications)

File: modules/ROOT/pages/changelogs.adoc (hunk 5 above)
Comment (Hunterness) [reply modules/ROOT/pages/changelogs.adoc:4]: went into 5.21 behind feature flag, so it's correct unless we want to wait for the feature flag to be removed to merge/introduce the notifications here

**Comment by Hunterness** [comment]
@renetapopova The PR introducing these notifications has been merged. However the feature is still behind a feature flag and not fully done, should we merge this already now or should it wait for the feature to be GA?

**Comment by renetapopova** [comment]
Is this notification visible to the user?

**Comment by Hunterness** [comment]
>Are these notifications visible to the user?

If you turn on the feature flag and try out the linked users feature and trigger them, yes. If you don't turn on the feature flag then no.

**Comment by renetapopova** [comment]
> > Are these notifications visible to the user?
> 
> If you turn on the feature flag (an internal setting) and try out the linked users feature and trigger them, yes. If you don't turn on the feature flag then no.

We don't surface internal settings, so I guess we don't need to merge it for now.

**Comment by renetapopova** [comment]
Related to https://github.com/neo4j/docs-operations/pull/1745

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-status-codes-142.surge.sh

Updated pages:
- [Changes to status codes per Neo4j version](https://neo4j-docs-status-codes-142.surge.sh/status-codes/5/changelogs)
- [List of all server notification codes](https://neo4j-docs-status-codes-142.surge.sh/status-codes/5/notifications/all-notifications)

**Comment by Hunterness** [comment]
Updated version tag to match the related docs PR

File: modules/ROOT/pages/changelogs.adoc (hunk 5 above)
Comment (Hunterness) [reply modules/ROOT/pages/changelogs.adoc]: Updated to 5.24 so it's the same as the main docs PR

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-status-codes-142.surge.sh

Updated pages:
- [Changes to status codes per Neo4j version](https://neo4j-docs-status-codes-142.surge.sh/status-codes/5/changelogs)
- [List of all server notification codes](https://neo4j-docs-status-codes-142.surge.sh/status-codes/5/notifications/all-notifications)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-status-codes-142.surge.sh

Updated pages:
- [Changes to status codes per Neo4j version](https://neo4j-docs-status-codes-142.surge.sh/status-codes/5/changelogs)
- [List of all server notification codes](https://neo4j-docs-status-codes-142.surge.sh/status-codes/5/notifications/all-notifications)

**Pull request merged**